### PR TITLE
Add support for multiple module per grade

### LINKED ISSUE #140: Update grade command to accept multiple module for the same index

### PR DESCRIPTION
**Pull request by yongqqqq**
### Description
This PR implements the improved grading feature for a student's module via a `GradeCommand`.

#### Main Changes:
- Updated `GradeCommand` to accept multiple module and grades per command.
- Updated `GradeCommandParser` to handle updated grade command input.
- Updated test case to match new module-grade pairing

#### Additional Changes:
- Refactoring to ensure code consistency
- Fix codestyle issues

#### Testing:
- All existing and new test cases pass
- This verifies that the updated grade command is being parsed and handled correctly.

closes #140

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/GradeCommand.java [hunk 1]
```diff
@@ -58,38 +61,43 @@ public CommandResult execute(Model model) throws CommandException {
...
+        Set<Module> personModules = new HashSet<>(personToGrade.getModules());
+        List<String> gradeResults = new ArrayList<>();
+
+        for (Map.Entry<String, Integer> entry : moduleGrades.entrySet()) {
+            String moduleName = entry.getKey();
+            int grade = entry.getValue();
+
+            if (grade < 0 || grade > 100) {
+                throw new CommandException(MESSAGE_INVALID_GRADE);
+            }
```
Comment (RadieonAjax) [inline src/main/java/seedu/address/logic/commands/GradeCommand.java:74]: Shouldn't be needed after the implementation of regex check.

**Review by RadieonAjax — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/GradeCommand.java (hunk 1 above)
Comment (yongqqqq) [reply src/main/java/seedu/address/logic/commands/GradeCommand.java]: removed the unnecessary check.

**Review by Jianwen0451 — APPROVED** [review]
LGTM! also, if you are using the ModelClassDiagram, I think we need to regenerate the image using the puml file. The png will not auto-update

**Pull request merged**